share/revng/test/configuration/revng-qa/compile.yml

```diff
@@ -1,27 +1,29 @@
 tags:
-  - name: clang
+  - name: clang-only
   - name: simple-executable
   - name: stripped
   - name: with-debug-info
 commands:
   - type: revng-qa.compiled-with-debug-info
     from:
       - type: source
-        filter: linux and simple-executable and !clang
+        filter: linux and simple-executable and !clang-only
     command: ${TRIPLE}gcc $INPUT -o $OUTPUT $GCC_CFLAGS $COMMON_CFLAGS
   - type: revng-qa.compiled-with-debug-info
     from:
       - type: source
-        filter: linux and simple-executable and clang
-    command: ${TRIPLE}clang $INPUT -o $OUTPUT $GCC_CFLAGS $COMMON_CFLAGS
+        filter: linux and simple-executable and clang-only
+    command: clang $INPUT -o $OUTPUT $GCC_CFLAGS $COMMON_CFLAGS $CLANG_CFLAGS
   - type: revng-qa.compiled-stripped
     from:
       - type: revng-qa.compiled-with-debug-info
+        filter: "!clang-only"
```

Comment:
1. This tag is being abused. If `strip` is ELF specific, let's have an `elf` tag and have `!elf`. Probably `!linux` is the right thing to have here.
2. Does `llvm-strip` support macho?

Reply:
1. Sounds good to me!

2. A quick google search shows that `llvm-objcopy --strip-all` is the way to go.

Reply:
1. Ok, this is _a lot_ more awkward than I expected. Thing is, all the `clang-only` architectures rely on others, so they end up inheriting `linux` (and `elf` too, if we were to add it). And we intentionally disallowed stripping tags. As such, the only viable way to proceed is to add a new tag :sadparrot: Or completely separate architectures. But that'd lead to duplication which IMO is even more dangerous.

2. Done.

Reply:
2. So can we drop using `${TRIPLE}strip` for `!clang-only` and `llvm-objcopy --strip-all` for `clang-only`? I still see both.

Let's review the usages of `clang-only`:
1. When we want to use `clang` (e.g., `share/revng/test/tests/analysis/Decompilation/clang/`). This can be the old `clang` tag.
2. For apple architectures (e.g., `apple-i386`, `apple-x86-64` and `apple-aarch64`) so we use cang. Those tags could simply imply `clang`.
3. In `revng-qa.runtime-abi-tests`/`revng-qa.runtime-abi-tests` to avoid running QEMU on those. We should use `!apple` there.

`clang-only` has a very ad-hoc, hacky and "excluding" semantic, I really want to avoid it.

Reply:
2. Wait, I think I lost you there. Do you mean we should get rid of `${TRIPLE}strip` and just use `llvm-objcopy --strip-all` for everything?

3. Wait, I'm confused about this too. Is it just the _name_ that you don't like? As in `clang-only`->`clang`?

3.3. What do you mean? We still want to run QEMU on everything it supports (which includes aarch64 apple binary).

Reply:
2. > Wait, I think I lost you there. Do you mean we should get rid of ${TRIPLE}strip and just use llvm-objcopy --strip-all for everything?

   Yes, any reason not to?
3. `clang` already existed and had a role which I think we want to preserve.

   > What do you mean? We still want to run QEMU on everything it supports (which includes aarch64 apple binary).
   
   My bad, I got confused.
   The problem is that in the first `revng-qa.runtime-abi-tests`  we use `!clang-only` and we use `prepare-elf-functions`. In the second instance we use `clang-only` + `prepare-macho-functions`. In the third we use `windows` and `prepare-pe-functions`.
   This should be reorganized so we have `linux` (or `elf`) for the first one, `apple` (or `macho`) for the second one and `windows` (or `pe-coff`) for the third one.
   Right now we have `!windows and !clang-only`, `!windows and clang-only` and then `windows`, very confusing.

I understand I'm no expressing myself too clearly, if you want I can give a shot at it.

Reply:
Done, please take a look!